agent/tools.go

```diff
@@ -17,5 +17,8 @@
 package tools
 
```

Comment:
Thanks for sharing all details on the PR overview! I was wondering if we should add a comment for this package to help people maintain tools/dependencies in the future.

```
// The tools package manages tool dependency with go mod.
// For tools that we need to generate code as part of `make gogenerate`, but are not automatically vendored, 
// we need to explicitly added them to this dummy tools.go file.
// 
// To vend a new tool, import the tool/package in this dummy tools.go file, and then rerun `go mod tidy` and `go mod vendor`. 
```

Reply:
Great idea, thanks for suggesting! I will take an action to add this in a future/followup PR.